### ISSUE TITLE
Flutter iOS distribution fails with No architectures in the binary

### ISSUE THREAD

**Issue by DeanPack** [flutter/flutter#101461]
## Steps to Reproduce

1. build a flutter app
2. install packages from my pubspec.yaml (listed below)
3. do a pod install
4. archive the build and try to upload to the appstore

**Expected results:** I expect the build to be uploaded

**Actual results:**after the archive finishes and I try to upload, I get an error:
ERROR ITMS-90085: "No architectures in the binary. Lipo failed to detect any architectures in the bundle executable."
Return status of iTunes Transporter was 1: ERROR ITMS-90085: "No architectures in the binary. Lipo failed to detect any architectures in the bundle executable."
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.

<details>
<summary>pubspec.yaml</summary>

```
name: app
description: A new Flutter project.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 2.0.1+1

environment:
  sdk: ">=2.15.0 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter
  shared_preferences: ^2.0.12
  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  firebase_crashlytics: ^2.6.0
  dots_indicator: ^2.1.0
  flutter_launcher_icons: ^0.9.2
  flutter_redux: ^0.9.0
  graphql: ^5.0.0
  jwt_decoder: ^2.0.1
  redux: ^5.0.0
  http: ^0.13.4
  webview_flutter: ^3.0.2
  mockito: ^5.0.17
  json_serializable: ^6.1.4
  json_annotation: ^4.4.0
  freezed: ^1.1.1
  freezed_annotation: ^1.1.0
  cached_network_image: ^3.2.0
  arrow_path: ^2.0.0
  
dev_dependencies:
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter
  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^1.0.0
  build_runner: ^2.1.7

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  assets: 
    - assets/


  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
```

</details>

<details>
<summary> podfile </summary>

```

# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

</details>

<details>
  <summary>Logs</summary>

The build runs with no errors, its only on ios distribution

```
```

```
dpackard@GX00060-MacBookPro app % flutter analyze
Analyzing app...                                                        
No issues found! (ran in 2.8s)
```

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.4, on macOS 11.6.4 20G417 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.59.1)
[✓] VS Code (version 1.57.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@DeanPack 
See [this](https://developer.apple.com/forums/thread/684969) similar issue reported in apple developer forum along with solution to update one of the pods OP was using. Please check the same at your end and see if it helps.

Another: https://developer.apple.com/forums/thread/665593, https://developer.apple.com/forums/thread/61454. Also check these out: https://github.com/OneSignal/OneSignal-iOS-SDK/issues/1052#issuecomment-1047780789


Maybe it has something to do with XCode 13.

**Comment by DeanPack** [comment]
@darshankawar Thanks for looking more into this for me! I thought the same, last night before I went to bed I went back to the last commit that I did a release for. I did it right after I created the demo project and added fastlane. I just went back to that commit, cleaned the build, and tried to release and it failed with the same error. The commit that the distribution worked on is all of the sudden failing, and that commit was before I added any pods to the project. The only thing I can think of that changed is that I'm running a newer flutter version now.
[✓] Flutter (Channel stable, 2.10.4, on macOS 11.6.4 20G417 darwin-x64, locale en-US)
